src/main/java/net/neoforged/neoforge/event/entity/player/PlayerRespawnPositionEvent.java

```diff
@@ -0,0 +1,157 @@
+        this.fromEndFight = fromEndFight;
+    }
+
+    /**
+     * @return The level the player will respawn into.
+     */
+    public ServerLevel getRespawnLevel() {
+        return respawnLevel;
+    }
+
+    /**
+     * Set the level the player will respawn into.
+     * 
+     * @param respawnLevel The new level.
+     */
+    public void setRespawnLevel(ServerLevel respawnLevel) {
+        this.respawnLevel = respawnLevel;
+    }
+
+    /**
+     * Set the level the player will respawn into using a {@link ResourceKey}.
+     * 
+     * @param respawnLevelResourceKey The {@link ResourceKey} of the level to respawn into.
+     */
+    public void setRespawnLevel(ResourceKey<Level> respawnLevelResourceKey) {
+        setRespawnLevel(Objects.requireNonNull(Objects.requireNonNull(getEntity().getServer()).getLevel(respawnLevelResourceKey)));
+    }
+
+    /**
+     * @return The level the server originally intended to respawn the player into.
+     */
+    public ServerLevel getOriginalRespawnLevel() {
+        return originalRespawnLevel;
+    }
+
+    /**
+     * @return The position in the target level where the player will respawn, before any adjustments by the server.
+     */
+    @Nullable
+    public Vec3 getRespawnPosition() {
+        return respawnPosition;
+    }
+
+    /**
+     * Set the player's respawn position within the respawn level. The server automatically adjusts this position
+     * to not be inside blocks. If {@code null}, the server will use the default spawn position for the level.
+     * 
+     * @param respawnPosition
+     */
+    public void setRespawnPosition(@Nullable Vec3 respawnPosition) {
+        this.respawnPosition = respawnPosition;
+    }
+
+    /**
+     * @return The original position the server intended to respawn the player at.
+     */
+    @Nullable
+    public Vec3 getOriginalRespawnPosition() {
+        return originalRespawnPosition;
+    }
+
+    /**
+     * @return The angle the player will face when they respawn, before any modifications made by the server.
+     */
+    public float getRespawnAngle() {
+        return respawnAngle;
+    }
+
+    /**
+     * Set the angle the player will face when they respawn. The server may adjust the angle, for example to face
+     * a bed if the player respawns there.
+     * 
+     * @param respawnAngle The angle the player will face when they respawn.
+     */
+    public void setRespawnAngle(float respawnAngle) {
+        this.respawnAngle = respawnAngle;
+    }
+
+    /**
+     * @return The original angle the server intended for the player to face when they respawn.
+     */
+    public float getOriginalRespawnAngle() {
+        return originalRespawnAngle;
+    }
+
+    /**
+     * @return Whether the respawn position will be used as the player's spawn position from then on. Defaults to {@code true}.
+     *         {@link PlayerSetSpawnEvent} will be fired if this is {@code true}.
+     */
+    public boolean isChangePlayerSpawnPosition() {
```

Comment:
The method name here fits better with a record-style getter `changePlayerSpawnPosition`

This is a stylistic thing, but `isChange` isn't grammatically correct.  The other option is `shouldChangeRespawnPos` (though the other method would need to be updated to match - `setShouldChangeRespawnPos`)